### ISSUE TITLE
onEnd called twice when paused prop changes at the end of the media

### ISSUE THREAD

**Issue by ldabiralai** [TheWidlarzGroup/react-native-video#2690]
# Bug

When using ExoPlayer, the onEnd prop is called for a second time when the paused prop changes in the `onEnd` first callback execution.

When the paused prop is updated using Android MediaPlayer the `onEnd` callback is not called for a second time.

## Platform
* Android ExoPlayer

## Environment info

React native info output:

```bash
System:
    OS: macOS 12.3.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 1.85 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.2 - ~/.volta/tools/image/node/14.18.2/bin/node
    Yarn: 1.22.18 - ~/.volta/tools/image/yarn/1.22.18/bin/yarn
    npm: 6.14.17 - ~/.volta/tools/image/npm/6.14.17/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 23, 29, 30, 31, 32
      Build Tools: 28.0.2, 28.0.3, 30.0.2, 31.0.0, 32.0.0
      System Images: android-28 | Google APIs Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom_64, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8139111
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.12 - /usr/local/opt/openjdk@11/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.68.2 => 0.68.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 5.2.0

## Steps To Reproduce

Run the reproducible code in linked, you can see the `onEnd` is logged to the console at the end of the video. Then switch to the `mediaplayer` branch in the repo and rerun `yarn android`, you can then see `onEnd` is only logged once.

## Expected behaviour

1. To be consistent with the MediaPlayer implementation I would expect `onEnd` to only be called once (at the end of the media playback) but on ExoPlayer it is being called twice.

## Reproducible sample code

https://github.com/ldabiralai/rnv-end-bug

## Video sample
n/a

**Comment by freeboub** [comment]
reproduced on master.

**Comment by freeboub** [comment]
and pull request opened

**Comment by freeboub** [comment]
merged for next release 6.0.0-alpha.2
